### PR TITLE
Adding presets

### PR DESCRIPTION
**Pull request by jeffreyvr**
This is still a work in progress, but I wanted to check with you (the maintainers 🙂) if it is worth to keep working on this.

This PR would add a command `php artisan forum:install {preset}`. Preset options would be `bootstrap` and `tailwind`. Along the lines of what has been said in https://github.com/Team-Tea-Time/laravel-forum/issues/337.

This command copies the `resources/views/forum` and `resources/components/forum` to the Laravel application. 

I have started working on converting all the views to Tailwind basically mimicking the bootstrap version, I think I'm at about 75%.

Would love to get feedback/thoughts and whether this is the way to go.

### CODE REVIEW

**Comment by Riari** [comment]
Hi - this looks great so far! It's pretty much what I had in mind as a first pass for UI presets - just a basic choice between the existing Bootstrap theme and a Tailwind version, both using Blade. Definitely keep working on it and I'll take a closer look ASAP. Thank you!

File: src/Console/Commands/InstallPreset.php [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+
+    public function handle()
+    {
+        $preset = $this->argument('preset', 'bootstrap');
+
+        (new Filesystem)->ensureDirectoryExists(resource_path('views/forum'));
+        (new Filesystem)->copyDirectory(__DIR__."/../../../stubs/{$preset}/resources/views", resource_path('views'));
+
+        (new Filesystem)->ensureDirectoryExists(resource_path('views/components/forum'));
+        (new Filesystem)->copyDirectory(__DIR__."/../../../stubs/{$preset}/resources/views/components/forum", resource_path('views/components'));
```
Comment (Riari) [inline src/Console/Commands/InstallPreset.php]: We can use DI here:
```suggestion
    public function handle(Filesystem $filesystem)
    {
        $preset = $this->argument('preset', 'bootstrap');

        $filesystem->ensureDirectoryExists(resource_path('views/forum'));
        $filesystem->copyDirectory(__DIR__."/../../../stubs/{$preset}/resources/views", resource_path('views'));

        $filesystem->ensureDirectoryExists(resource_path('views/components/forum'));
        $filesystem->copyDirectory(__DIR__."/../../../stubs/{$preset}/resources/views/components/forum", resource_path('views/components'));
```

File: src/Console/Commands/InstallPreset.php [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+
+use Illuminate\Console\Command;
+use Illuminate\Filesystem\Filesystem;
+use Symfony\Component\Finder\Finder;
+
+class InstallPreset extends Command
+{
+    protected $signature = 'forum:install {preset}';
+
+    protected $description = 'Install the forum preset.';
```
Comment (Riari) [inline src/Console/Commands/InstallPreset.php]:
```suggestion
    protected $description = 'Install a UI preset for the forum.';
```

File: src/Http/Controllers/Api/BaseController.php [hunk 3]
```diff
@@ -27,7 +27,7 @@ protected function bulkActionResponse(int $rowsAffected, string $transKey): Resp
         return new Response([
             'success' => true,
             'rows_affected' => $rowsAffected,
-            'message' => trans_choice("forum::{$transKey}", $rowsAffected),
+            'message' => trans_choice("forum.{$transKey}", $rowsAffected),
```
Comment (Riari) [inline src/Http/Controllers/Api/BaseController.php]: Just for my own information - what's the reason for this change? Is it due to a change in how Laravel handles paths?

**Review by Riari — COMMENTED** [review]
I've left some feedback for the changes so far, but they are looking pretty good overall. Really appreciate the effort!

File: src/Http/Controllers/Api/BaseController.php (hunk 3 above)
Comment (jeffreyvr) [reply src/Http/Controllers/Api/BaseController.php]: That change was accidental. Has been fixed.

**Comment by jeffreyvr** [comment]
I think it is now at a pretty good stage. If you run into anything on review, let me know :)

**Comment by Riari** [comment]
@jeffreyvr Really sorry I took so long to get back to you! This looks great. I just pushed some small fixes/cleanup. I'm pretty sure this is the biggest feature ever contributed to the package, so thank you for that. I'll get this released and update the docs.

**Comment by jeffreyvr** [comment]
No problem at all! Happy to contribute and thanks for the fixes/clean ups (and merge) :-)
